test/filtering-huge-grid.html

```diff
@@ -0,0 +1,122 @@
+<!doctype html>
+
+<html>
+
+<head>
+  <meta charset="UTF-8">
+  <title>Filtering huge grid test</title>
+  <meta name="viewport" content="width=device-width, initial-scale=1.0, maximum-scale=1.0">
+
+  <script src="../../webcomponentsjs/webcomponents-lite.js"></script>
+  <script src="../../web-component-tester/browser.js"></script>
+  <link rel="import" href="../../test-fixture/test-fixture.html">
+
+  <link rel="import" href="helpers.html">
+
+  <link rel="import" href="../vaadin-grid.html">
+  <link rel="import" href="../vaadin-grid-column.html">
+  <link rel="import" href="../vaadin-grid-filter.html">
+</head>
+
+<body>
+
+  <dom-module id="x-grid">
+    <template>
+      <style>
+        .item {
+          height: 30px;
+        }
+
+        vaadin-grid-cell-content {
+          /* Override Lumo theme */
+          padding: 0 !important;
+        }
+      </style>
+      <vaadin-grid id="grid" style="height: 300px;">
+        <vaadin-grid-column>
+          <template class="header">
+            <div>Value</div>
+            <vaadin-grid-filter path="value">
+            </vaadin-grid-filter>
+          </template>
+          <template>[[item.value]]</template>
+        </vaadin-grid-column>
+      </vaadin-grid>
+    </template>
+    <script>
+      customElements.whenDefined('vaadin-grid').then(() => {
+        Polymer({
+          is: 'x-grid'
+        });
+      });
+    </script>
+  </dom-module>
+
+  <test-fixture id="table">
+    <template>
+      <x-grid></x-grid>
+    </template>
+  </test-fixture>
+
+  <script>
+
+    describe('filtering', function() {
+      let container, grid, filter;
+      const maxgridsize = 300000;
+      this.timeout(3000000);
+
+      before(done => ensureDefined('x-grid', done));
+
+      describe('items', () => {
+        beforeEach(done => {
+          container = fixture('table');
+          grid = container.$.grid;
+          grid.size = maxgridsize;
+          grid.dataProvider = (params, callback) => {
+            let numberOfElements = maxgridsize;
+            params.filters.forEach(function(filter) {
+              if (isNaN(filter.value)) {
+                numberOfElements = maxgridsize;
+              } else {
+                numberOfElements = filter.value;
+              }
+            });
+            grid.size = numberOfElements;
+            var response = [];
+            for (var i = 0; i <= params.pageSize; i++) {
+              var element = params.page * params.pageSize + i;
+              if (element < grid.size) {
+                response.push({value: 'item' + element});
+              }
+            }
+            callback(response);
+          };
+          expect(getCellContent(getFirstVisibleItem(grid)).textContent).to.contain('item0');
```

Comment:
Why is `expect` used outside of the test below, in `beforeEach` block?

When I remove this line, I'm getting the following error:

```
Error: expected 'item0' to include 'item299975'
```

Reply:
I removed it and changed the test. It was wrong (and causes the timeout).